src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -418,11 +418,9 @@ QString QgsOracleUtils::whereClause( QgsFeatureId featureId, const QgsFields &fi
     case PktRowId:
     case PktFidMap:
     {
-      QVariant pkValsVariant = sharedData->lookupKey( featureId );
-      if ( !pkValsVariant.isNull() )
+      QVariantList pkVals = sharedData->lookupKey( featureId );
```

Comment:
What's actually happening here? Is a non-list value getting added to the lookup somehow?

Reply:
No, the lookup map is empty, and returns en [empty QVariantList](https://github.com/troopa81/QGIS/blob/fix_wcs_blank_tile/src/providers/oracle/qgsoracleprovider.cpp#L3381) for the given featureId. Empty list is not a null variant, so the condition needs to be changed (it's pretty much the same behavior in postgres provider)

Reply:
Ah, good catch

Reply:
Looks like the postgres provider already has this fix too